tests/test_elastigroup.py

```diff
@@ -167,3 +168,29 @@ def test_get_elastigroup_instances(data, err, result, expected):
         elastigroup_status_mock.side_effect = err
         got = zmon_aws_agent.elastigroup.get_elastigroup_instances(data)
         assert got == expected
+
+
+def test_extract_instance_details():
+    # Example from https://api.spotinst.com/spotinst-api/elastigroup/amazon-web-services/status/
+    resp = '{"request":{"id":"890a90c2-5264-482b-a72b-e021557227e4","url":"/aws/ec2/group/sig-12345678/status",' \
+           '"method":"GET","timestamp":"2018-06-25T11:51:42.629Z"},"response":{"status":{"code":200,"message":"OK"},' \
+           '"kind":"spotinst:aws:ec2:group","items":[{"spotInstanceRequestId":"sir-3thgagpn",' \
+           '"instanceId":"i-0cc289f12538e4758","instanceType":"t2.micro","product":"Linux/UNIX",' \
+           '"groupId":"sig-12345678","availabilityZone":"us-west-2a","privateIp":"172.31.28.210",' \
+           '"createdAt":"2018-06-25T11:49:00.000Z","publicIp":"10.10.10.10","status":"fulfilled"},' \
+           '{"spotInstanceRequestId":null,"instanceId":"i-05ebb28abebdc718b","instanceType":"t2.medium",' \
+           '"product":"Linux/UNIX","groupId":"sig-05417358","availabilityZone":"us-west-2a",' \
+           '"privateIp":"172.31.17.189","createdAt":"2018-06-25T11:49:02.000Z","publicIp":"10.10.10.10",' \
+           '"status":"running"}],"count":2}}'
+    with requests_mock.Mocker() as m:
+        m.get("https://api.spotinst.io/aws/ec2/group/42/status", text=resp)
```

Comment:
We can use `json=resp` instead of having a string response. And convert `resp` to be a dict for better readability.

Reply:
This test was deliberately created like this just to test how the spotinst SDK would handle the `null` value in one of the examples. If I provided a dict I'd be skipping that relevant part of the test.

Reply:
Python's `HEREDOC` isn't so elegant as Ruby's one, but you still may use it here for readability sake.

```python
import re

resp = re.sub("\s+", "", '''\
{\
  "request":{\
    "id":"890a90c2-5264-482b-a72b-e021557227e4",\
    "url":"/aws/ec2/group/sig-12345678/status",\
    "method":"GET",\
    "timestamp":"2018-06-25T11:51:42.629Z"},\
    "response":{\
      "status":{\
        "code":200,\
        "message":"OK"\
      },\
      "kind":"spotinst:aws:ec2:group",\
      "items":[{\
        "spotInstanceRequestId":"sir-3thgagpn",\
        "instanceId":"i-0cc289f12538e4758",\
        "instanceType":"t2.micro",\
        "product":"Linux/UNIX",\
        "groupId":"sig-12345678",\
        "availabilityZone":"us-west-2a",\
        "privateIp":"172.31.28.210",\
        "createdAt":"2018-06-25T11:49:00.000Z",\
        "publicIp":"10.10.10.10",\
        "status":"fulfilled"\
      }, {\
        "spotInstanceRequestId":null,\
        "instanceId":"i-05ebb28abebdc718b",\
        "instanceType":"t2.medium",\
        "product":"Linux/UNIX",\
        "groupId":"sig-05417358",\
        "availabilityZone":"us-west-2a",\
        "privateIp":"172.31.17.189",\
        "createdAt":"2018-06-25T11:49:02.000Z",\
        "publicIp":"10.10.10.10",\
        "status":"running"\
      }],\
    "count":2\
  }\
}\
''')
```

`\` is used to prevent newlines. Plus you can use a bit of post-processing to get rid of formatting spaces.

In addition if you want to allow empty spaces inside double quotes you can use look behind and look ahead feature of regex to avoid space replacement in quotes.